agent/agent.go

```diff
@@ -241,15 +241,43 @@ func (a *Agent) Start() error {
 		return fmt.Errorf("Failed to setup node ID: %v", err)
 	}
 
+	// create the config for the rpc server/client
+	consulCfg, err := a.consulConfig()
+	if err != nil {
+		return err
+	}
+
 	// Setup either the client or the server.
 	if c.Server {
-		server, err := a.makeServer()
+		serverUp := make(chan int)
+		consulCfg.ServerUp = func() { close(serverUp) }
+
+		server, err := consul.NewServerLogger(consulCfg, a.logger)
 		if err != nil {
-			return err
+			return fmt.Errorf("Failed to start Consul server: %v", err)
 		}
 
 		a.delegate = server
-		a.state.Init(c, a.logger, server)
+
+		// wait for the server to be ready
+		// then trigger an anti-entropy run ???
+
+		// this needs review since there is no for the state to be
```

Comment:
I think a better way to solve this is to pass a channel between the two components. They're thread-safe and were added to the language for that purpose. I'll create it with a reasonable buffer so that we don't need the non-blocking select unless that is a reasonable concern.